### ISSUE TITLE
Internet-connected receivers with duplicate serial numbers

### ISSUE THREAD

**Issue by StuMackenzie** [MotusDev/Motus-TO-DO#502]
This has been dealt before in one way or another, but there is still be a missing link somewhere.

Receiver SG-2616BBBK1111_1 is online and setup at Western U. However, it is being listed as Loring BIWF on the sensorgnome status page - https://sensorgnome.org/status/ 

Also see

see - https://github.com/MotusDev/Motus-TO-DO/issues/304

**Comment by Hussell** [comment]
Possibly related: https://sgdata.motus.org/status?jobID=734769

From Stu: This was the receiver 2616BBBK1111_1 that is at Western. Is it really returning no records, or is it related to the duplicate issue? There was data recorded for every day of the period.

**Comment by StuMackenzie** [comment]
The last upload from this station was ok Dec 2018. However it has been online and probably uploading to the other deployment with Sag-2616BBBK1111 - so I suppose the question is whether the auto-upload takes duplicate serials into account?

____________________

On Jun 7, 2019, at 09:17, Jeremy Hussell <notifications@github.com<mailto:notifications@github.com>> wrote:


Possibly related: https://sgdata.motus.org/status?jobID=734769

From Stu: This was the receiver 2616BBBK1111_1 that is at Western. Is it really returning no records, or is it related to the duplicate issue? There was data recorded for every day of the period.

-

**Comment by Hussell** [comment]
That's it. The syncReceiver jobs have been putting the files into the wrong repositories. The detections appear to be going to the correct receiver in the database though.

**Comment by cjardine-bsc** [comment]
Okay I’m looking back into this. 
 
For reference: 
Deviceid 1017    receiverID SG-2616BBBK1111 ProjectID 14
 
Deviceid 1077    receiverID SG-2616BBBK1111_1 ProjectID 1 
 
Currently all the batches for that serial number are being assigned to deviceID  1077 (Project 1) 
 
The last time any data was assigned to 1017 (project 14) was 2018-08-01 14:43:45.050 
 
There are a bunch of batches that have come in for 1077 in June. But they appear problematic. They have the exact same monoBN, start and end times and zero hits. Which makes me think it’s an issue with the receiver packaging its detection files before sending to tagfinder (totally guessing). 
 
Over to you Jeremy. 
 
 
batchID	projectID	motusDeviceID	(No column name)	(No column name)	monoBN	tsBegin	tsEnd	numHits	motusUserID	motusJobID	ts	tsMotus	batchFilter	version
441078	NULL	1077	1/1/70 0:00	6/13/19 0:09	24	0	1560384567	0	347	775553	1.56E+09	1.56E+09	0	1
441114	NULL	1077	6/13/19 1:09	6/13/19 1:09	24	1560388185	1560388185	0	347	775777	1.56E+09	1.56E+09	0	1
441157	NULL	1077	6/13/19 2:10	6/13/19 2:10	24	1560391803	1560391803	0	347	776031	1.56E+09	1.56E+09	0	1
441220	NULL	1077	6/13/19 3:10	6/13/19 4:10	24	1560395412	1560399029	0	347	776398	1.56E+09	1.56E+09	0	1
441274	NULL	1077	6/13/19 5:10	6/13/19 5:10	24	1560402630	1560402630	0	347	776743	1.56E+09	1.56E+09	0	1
441327	NULL	1077	6/13/19 6:10	6/13/19 6:10	24	1560406234	1560406234	0	347	777025	1.56E+09	1.56E+09	0	1
441377	NULL	1077	6/13/19 7:10	6/13/19 7:10	24	1560409843	1560409843	0	347	777333	1.56E+09	1.56E+09	0	1
441431	NULL	1077	6/13/19 8:10	6/13/19 8:10	24	1560413451	1560413451	0	347	777573	1.56E+09	1.56E+09	0	1
441481	NULL	1077	6/13/19 9:10	6/13/19 9:10	24	1560417053	1560417053	0	347	777876	1.56E+09	1.56E+09	0	1
441517	NULL	1077	6/13/19 10:10	6/13/19 10:10	24	1560420654	1560420654	0	347	778083	1.56E+09	1.56E+09	0	1
441572	NULL	1077	6/13/19 11:10	6/13/19 11:10	24	1560424255	1560424255	0	347	778415	1.56E+09	1.56E+09	0	1
441626	NULL	1077	6/13/19 12:10	6/13/19 12:10	24	1560427858	1560427858	0	347	778741	1.56E+09	1.56E+09	0	1
441738	NULL	1077	6/13/19 13:10	6/13/19 13:10	24	1560431459	1560431459	0	347	779063	1.56E+09	1.56E+09	0	1
441767	NULL	1077	6/13/19 14:11	6/13/19 14:11	24	1560435060	1560435060	0	347	779238	1.56E+09	1.56E+09	0	1
441869	NULL	1077	6/13/19 15:11	6/13/19 15:11	24	1560438661	1560438661	0	347	779705	1.56E+09	1.56E+09	0	1
441915	NULL	1077	6/13/19 16:11	6/13/19 16:11	24	1560442263	1560442263	0	347	779977	1.56E+09	1.56E+09	0	1
441979	NULL	1077	6/13/19 17:11	6/13/19 18:11	24	1560445864	1560449474	0	347	780361	1.56E+09	1.56E+09	0	1
442034	NULL	1077	6/13/19 19:11	6/13/19 19:11	24	1560453076	1560453076	0	347	780691	1.56E+09	1.56E+09	0	1
442091	NULL	1077	6/13/19 20:11	6/13/19 20:11	24	1560456684	1560456684	0	347	781030	1.56E+09	1.56E+09	0	1
442142	NULL	1077	6/13/19 21:11	6/13/19 21:11	24	1560460303	1560460303	0	347	781341	1.56E+09	1.56E+09	0	1

**Comment by cjardine-bsc** [comment]
also note that none of those batches have GPS data.

**Comment by Hussell** [comment]
The monoBN being the same is normal for a synched receiver: a new batch is being sent every hour, but the receiver doesn't get rebooted each time, so each new hour's worth of data is just added to the same boot session.

Missing GPS and tsEnd being identical to tsBegin is a problem. That's probably caused by the files going into the wrong repository on sgdata, then the tagfinder using the correct serial number to look for files in the correct repository (where the files aren't).

**Comment by Hussell** [comment]
I've created a fix which (should) make the syncServer add the suffix to duplicate serial numbers as early as possible. Once that's reviewed, running and confirmed to be working, I think the only thing I'll have to do is move all the files in the wrong repository back to where they should be, then rerun the receiver.

**Comment by Hussell** [comment]
Ok, I think the changes I've made are dealing with this receiver correctly now. There's about a day's worth of activity now, starting yesterday. I've moved all the files to the correct places in the repositories, so the only thing that remains is to do a full rerun of both receivers. Catherine, I see that receiver SG-2616BBBK1111 (device ID 1017) currently has batches which belong to SG-2616BBBK1111_1 (device ID 1077), e.g. everything before 2017-07-11 except for 2017-06-12, plus others. Is it possible to delete or archive every existing batch for these two receivers just before I rerun everything? That's probably the surest way to fix the current situation. Maybe move them all to SG-2616BBBK1111_2 (device ID 1219)?

**Comment by cjardine-bsc** [comment]
I'm confused. I thought we had established that these batches were being correctly assigned to the right receiver on our end?

**Comment by Hussell** [comment]
These are probably batches left over from before #220. It looks as though we never cleaned up (some of?) the batches from 2017-2018.

**Comment by cjardine-bsc** [comment]
Okay, go ahead and re-run everything. I'll correct any remaining errors in batchID once that is done. The existing batches won't impact the re-run.

**Comment by Hussell** [comment]
Ok, both receivers have been rerun and the batches have been imported into the database. Now there's just getting rid of the old messed-up batches. E.g. 451233, which combines data from the two receivers thanks to the monoBN issue.

**Comment by denislepage** [comment]
Also remember that we will be re-importing the data again for the new tables shortly. If there are customizations that are not part of the automated processes, we’ll need to take that into account.

Make sure to keep track of what fields/tables those are affecting.

**Comment by StuMackenzie** [comment]
SG-2616BBBK1111 - this receiver is still showing up as Loring BIWF even though all deployments in that project (14) are ended. Deployment under Project 1 - UWO is active (5066).

**Comment by cjardine-bsc** [comment]
Jeremy, to confirm, all batches for deviceID 1077 and 1017 run prior to Jun 26 2019 should be set as duplicates correct? I.e. batches with sgBatchID < 453559 

SELECT *  ,dbo.ts2datetime([tsMotus])
  FROM [motus].[dbo].[sg_batches]
  where motusDeviceID in (1077, 1017)
  order by batchID

**Comment by Hussell** [comment]
Correct.

**Comment by cjardine-bsc** [comment]
Okay I've run that on the sg_batches and det_batches tables in live.

I've mode a note in prSGBatches_filters_ts2 with the script in case we need to run it again. The data will still be in the detection tables until we implement the code to delete deprecated batches.
